src/system.h

```diff
@@ -422,6 +422,21 @@ void Panic_(const char * file, int line, const char * fmt, ...) NORETURN
 #define Panic(...) \
     Panic_(__FILE__, __LINE__, __VA_ARGS__)
 
+void OutOfMemoryPanic(UInt size, const char * reason);
+
+/****************************************************************************
+**
+*F SetTooLargeAllocationHandler
```

Comment:
```suggestion
*F  SetTooLargeAllocationHandler( handler )
```

Reply:
In any case, I wonder: Do we really need this `SetTooLargeAllocationHandler` stuff at all? It seems like generalization just for its own sake. Without it, this PR becomes substantially simpler, and I can't think of any applications for it right now. At the same time, it's one more API that in principle now *all* our GCs would have to support (this PR only deals with GASMAN).